GuruLoader/GuruLoader.cs

```diff
@@ -269,5 +334,32 @@ public static string ComposeGuruUrl(string cik)
             return CreateDisplayPortfolio(rssData.DisplayName, portfolios[0], portfolios[1]);
         }
     }
+
+    public async static Task<FullPortfolioData> FetchFullPortfolioData(string cik) {
+        if (String.IsNullOrEmpty(cik)) throw new Exception("Cik cannot be empty");
+
+        var rssUrl = ComposeGuruUrl(cik);
+        using (var client = new HttpClient()) {
+            // Getting the rss stream cannot be started in parallel as it needs to be read before loading the portfolios
+            var rss = await client.GetStreamAsync(rssUrl);
+            var rssData = ParseRssText(rss);
+
+            var portsNumber = rssData.Links.Count();
+            if (portsNumber == 0) throw new Exception("No portfolios for this cik");
+
+            var ports = rssData.Links.Select(l => FetchPortfolio(client, l));
```

Comment:
This method is very similar to FetchDisplayPortfolio. If you add a .First where you get the Links property here, they'd be almost the same. I wonder if you could combine them into a single method that takes an optional integer parameter specifying how many portfolios back to look.